File: ingest-rest/src/main/java/org/chronopolis/ingest/api/serializer/ZonedDateTimeSerializer.java
package org.chronopolis.ingest.api.serializer;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;

import java.io.IOException;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;
import java.time.format.DateTimeFormatter;

/**
 * Jackson serializer for ZonedDateTime objects
 *
 * Created by shake on 12/16/16.
 */
public class ZonedDateTimeSerializer extends JsonSerializer<ZonedDateTime> {
    @Override
    public void serialize(ZonedDateTime localDateTime,
                          JsonGenerator jsonGenerator,
                          SerializerProvider serializerProvider) throws IOException {
        DateTimeFormatter fmt = DateTimeFormatter.ISO_OFFSET_DATE_TIME.withZone(ZoneOffset.UTC);
        jsonGenerator.writeString(fmt.format(localDateTime));
    }
}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/entities/serializers/ZonedDateTimeSerializer.kt
package org.chronopolis.rest.entities.serializers

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import java.time.ZoneOffset.UTC
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME

/**
 * Simple serializer for writing [ZonedDateTime] as an ISO-8601 string at UTC-0
 *
 * @author shake
 */
@Deprecated("already exists")
class ZonedDateTimeSerializer : JsonSerializer<ZonedDateTime>() {
    override fun serialize(value: ZonedDateTime,
                           gen: JsonGenerator,
                           serializers: SerializerProvider) {
        gen.writeString(ISO_OFFSET_DATE_TIME.withZone(UTC).format(value))
    }
}